internal/btf/btf.go

```diff
@@ -107,9 +107,6 @@ func LoadSpecFromReader(rd io.ReaderAt) (*Spec, error) {
 		}
 
 		secName := file.Sections[symbol.Section].Name
-		if symbol.Value > math.MaxUint32 {
-			return nil, fmt.Errorf("section %s: symbol %s: size exceeds maximum", secName, symbol.Name)
-		}
 
 		variableOffsets[variable{secName, symbol.Name}] = uint32(symbol.Value)
```

Comment:
In that case variable offsets should also be 64 bit? With your change we'd just silently truncate.

Reply:
Yep was wondering about this. As a datapoint, at least on the vmlinux I have here the symbol addresses are either of the form `ffffffffXXXXXXXX`or `00000000XXXXXXXX`: 
```
155662: ffffffff81968bc0    23 FUNC    GLOBAL DEFAULT     1 dm_target_exit
155663: ffffffff831546b8     0 NOTYPE  GLOBAL DEFAULT    30 __stop___trace_bprintk_fmt
155664: 000000009aaeefce     0 NOTYPE  GLOBAL DEFAULT   ABS __crc_sysctl_nf_log_all_netns
155665: 00000000949b5176     0 NOTYPE  GLOBAL DEFAULT   ABS __crc_devlink_region_snapshot_create
155666: ffffffff810b5450   143 FUNC    GLOBAL DEFAULT     1 __x64_sys_ssetmask
155667: 000000004456717a     0 NOTYPE  GLOBAL DEFAULT   ABS __crc_devlink_trap_policers_register
155668: ffffffff831004e0   432 OBJECT  GLOBAL DEFAULT    30 ping_prot
155669: ffffffff81703420    33 FUNC    GLOBAL DEFAULT     1 clk_hw_unregister_gate
155670: ffffffff81317320   599 FUNC    GLOBAL DEFAULT     1 zbud_reclaim_page
```

Whether that means we can ignore the first 32 bits or not, I don't know :shrug:

Reply:
Okay, taking another look at this, this is indeed not a good idea. Truncation would cause symbols `ffffffff81317320` and `0000000081317320` to have the same offset. Not an issue in practice, because:
- symbol values aren't used as an index, and lookups are always performed based on sec/symbol name.
- BTF datasec vars have 32-bit offsets, so symbols sitting in the `ffffffff00000000` range simply cannot be encoded

My vmlinux seems to confirm this. There is only one datasec:
```
~ bpftool btf dump file /usr/lib/modules/5.16.2-zen1-1-zen/build/vmlinux
...
[107260] DATASEC '.data..percpu' size=209156 vlen=380
        type_id=5390 offset=4096 size=4096 (VAR 'cpu_debug_store')
...
```

```
~ llvm-readelf -s /usr/lib/modules/5.16.2-zen1-1-zen/build/vmlinux | grep cpu_debug_store
...
142762: 0000000000001000  4096 OBJECT  GLOBAL DEFAULT    37 cpu_debug_store
```

Since vmlinux `.text` sits in `ffffffff81000000` (and is not a datasec), there cannot be any BTF references to it.

Because of all this, I think the straightforward solution would be to skip any symbols whose addresses overflow u32, since they cannot be referenced by BTF.